[improve][doc] SEO for Concepts and Architecture except Overview and Messaging (#674)
apache/pulsar-site: docs/concepts-cluster-level-failover.md

id: concepts-cluster-level-failover
title: Cluster-level failover
sidebar_label: "Cluster-level failover"

> description: Get a comprehensive understanding of the concept, benefits, use cases and more information about the cluster-level failover in Pulsar.

Reviewer: ```suggestion
description: Get a comprehensive understanding of concepts, benefits, and use cases about the cluster-level failover in Pulsar.
```

The cluster-level failover provides fault tolerance, continuous availability, and high availability together. It brings a number of benefits, including but not limited to:

* Reduced cost

   services can be switched and recovered automatically with no data loss.

* Simplified management

   businesses can operate on an "always-on" basis since no immediate user intervention is required.

* Improved stability and robustness

>    it ensures continuous performance and minimizes service downtime.

Reviewer: Why not capitalize the 1st letter of the 1st word in a sentence?

### When to use cluster-level failover?

The cluster-level failover protects your environment in a number of ways, including but not limited to:

* Disaster recovery
   
   cluster-level failover can automatically and seamlessly transfer the production workload on a primary cluster to one or several backup clusters, which ensures minimum data loss and reduced recovery time.

* Planned migration
   

>    if you want to migrate production workloads from an old cluster to a new cluster, you can improve the migration efficiency with cluster-level failover. For example, you can test whether the data migration goes smoothly in case of a failover event, identify possible issues and risks before the migration.

Reviewer: same question as above

Automatic cluster-level failover is triggered when Pulsar clients cannot connect to the primary cluster for a prolonged period of time. This can be caused by any number of reasons including, but not limited to:

* Network failure
   
   internet connection is lost.

* Power failure

   shutdown time of a primary cluster exceeds time limits.

* Service error

   errors occur on a primary cluster (for example, the primary cluster does not function because of time limits).

* Crashed storage space

>    the primary cluster does not have enough storage space, but the corresponding storage space on the backup server functions normally.

Reviewer: same

### Why does cluster-level failover fail?

> Obviously, the cluster-level failover will fail if the backup cluster is unreachable by active Pulsar clients. This can happen for many reasons, including but not limited to:

Reviewer: ```suggestion
Obviously, the cluster-level failover does not succeed if the backup cluster is unreachable by active Pulsar clients. This can happen for many reasons, including but not limited to:
```

1. Avoid duplicate "fail"
2. Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true. https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp

* Power failure

   the backup cluster is shut down or does not function normally.

* Crashed storage space

>    primary and backup clusters do not have enough storage space.

Reviewer: same

   3b) If the primary cluster does not come back, the Pulsar client does not perform the switchover.

![Workflow of automatic failover cluster in Pulsar](/assets/cluster-level-failover-4.png)

</TabItem>
<TabItem value="Controlled cluster-level failover">

> The controlled failover cluster performs the following actions with administrator intervention:

Reviewer: Good catch!
